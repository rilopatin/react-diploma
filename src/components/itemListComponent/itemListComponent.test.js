import React from 'react'
import ItemListComponent from './itemListComponent'
import {shallow} from "enzyme/build/index";
import ErrorMessage from '../errorMessage/errorMessage'

describe('Testing <ItemListComponent/>', () => {

    let items = [{name: 'wqw1', country: 'Brazil'}, {name: 'wqw2', country: 'Columbia'}, {name: 'sdgf', country: 'Brazil'}]
    const tempFunc = async () => {
        return await items
    }
        const wrapper = shallow(<ItemListComponent classes={["shop__item", "shop__item-title", "shop__item-price", "shop__wrapper"]}
                                                   getData = {tempFunc}
                                                   renderItem={() => null}
                                                   term = ''
                                                   filter = ''
                                                   details
        />)

    it('ItemListComponent has been rendered correctly', () => {
        expect(wrapper).toMatchSnapshot()
    });

    it('List of 3 items is rendered to 3 instance', () => {
        expect(wrapper.find('li')).toHaveLength(3)
    });

    it('List of 3 items with filter is rendered to 2 instance', () => {
        wrapper.setProps({filter:  'Brazil'})
        expect(wrapper.find('li')).toHaveLength(2)
    });

    it('List of 3 items with filter and term is rendered to 1 instance', () => {
        wrapper.setProps({term:  'wqw'})
        expect(wrapper.find('li')).toHaveLength(1)
    });

    it('ItemListComponent state "error" is false', () => {
        expect(wrapper.state.error).toBeFalsy()
    });

    describe('Handlers tests', () => {
        it('testing onError', () => {
            wrapper.instance().onError(<ErrorMessage code={'404'}/>);
            expect(wrapper.state().loading).toBeFalsy();
            expect(wrapper.state().error).toBeTruthy()
        });
        it('testing onItemSelected', () => {
            wrapper.instance().onItemSelected('1')
            expect(wrapper.state().selectedItem).toEqual('1')
        })

    })
})
