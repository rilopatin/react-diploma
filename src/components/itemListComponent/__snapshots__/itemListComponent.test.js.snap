// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ItemListComponent/> ItemListComponent has been rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemListComponent
    classes={
      Array [
        "shop__item",
        "shop__item-title",
        "shop__item-price",
        "shop__wrapper",
      ]
    }
    details={true}
    filter=""
    getData={[Function]}
    renderItem={[Function]}
    term=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        null,
        Array [
          <li
            className="shop__item"
            onClick={[Function]}
          >
            <Link
              replace={false}
              to="/details/wqw1"
            >
              <Img
                alt="coffee"
                container={[Function]}
                decode={true}
                loader={false}
                src={Array []}
                unloader={false}
              />
              <div
                className="shop__item-title"
              >
                wqw1
              </div>
              <div
                className="shop__item-country"
              />
              <div
                className="shop__item-price"
              />
            </Link>
          </li>,
          <li
            className="shop__item"
            onClick={[Function]}
          >
            <Link
              replace={false}
              to="/details/wqw2"
            >
              <Img
                alt="coffee"
                container={[Function]}
                decode={true}
                loader={false}
                src={Array []}
                unloader={false}
              />
              <div
                className="shop__item-title"
              >
                wqw2
              </div>
              <div
                className="shop__item-country"
              />
              <div
                className="shop__item-price"
              />
            </Link>
          </li>,
          <li
            className="shop__item"
            onClick={[Function]}
          >
            <Link
              replace={false}
              to="/details/sdgf"
            >
              <Img
                alt="coffee"
                container={[Function]}
                decode={true}
                loader={false}
                src={Array []}
                unloader={false}
              />
              <div
                className="shop__item-title"
              >
                sdgf
              </div>
              <div
                className="shop__item-country"
              />
              <div
                className="shop__item-price"
              />
            </Link>
          </li>,
        ],
      ],
      "className": "shop__wrapper",
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
      Object {
        "instance": null,
        "key": "wqw1",
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/details/wqw1"
          >
            <Img
              alt="coffee"
              container={[Function]}
              decode={true}
              loader={false}
              src={Array []}
              unloader={false}
            />
            <div
              className="shop__item-title"
            >
              wqw1
            </div>
            <div
              className="shop__item-country"
            />
            <div
              className="shop__item-price"
            />
          </Link>,
          "className": "shop__item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Img
                alt="coffee"
                container={[Function]}
                decode={true}
                loader={false}
                src={Array []}
                unloader={false}
              />,
              <div
                className="shop__item-title"
              >
                wqw1
              </div>,
              <div
                className="shop__item-country"
              />,
              <div
                className="shop__item-price"
              />,
            ],
            "replace": false,
            "to": "/details/wqw1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "coffee",
                "container": [Function],
                "decode": true,
                "loader": false,
                "src": Array [],
                "unloader": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "wqw1",
                "className": "shop__item-title",
              },
              "ref": null,
              "rendered": "wqw1",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
                "className": "shop__item-country",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "shop__item-price",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "wqw2",
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/details/wqw2"
          >
            <Img
              alt="coffee"
              container={[Function]}
              decode={true}
              loader={false}
              src={Array []}
              unloader={false}
            />
            <div
              className="shop__item-title"
            >
              wqw2
            </div>
            <div
              className="shop__item-country"
            />
            <div
              className="shop__item-price"
            />
          </Link>,
          "className": "shop__item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Img
                alt="coffee"
                container={[Function]}
                decode={true}
                loader={false}
                src={Array []}
                unloader={false}
              />,
              <div
                className="shop__item-title"
              >
                wqw2
              </div>,
              <div
                className="shop__item-country"
              />,
              <div
                className="shop__item-price"
              />,
            ],
            "replace": false,
            "to": "/details/wqw2",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "coffee",
                "container": [Function],
                "decode": true,
                "loader": false,
                "src": Array [],
                "unloader": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "wqw2",
                "className": "shop__item-title",
              },
              "ref": null,
              "rendered": "wqw2",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
                "className": "shop__item-country",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "shop__item-price",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "sdgf",
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/details/sdgf"
          >
            <Img
              alt="coffee"
              container={[Function]}
              decode={true}
              loader={false}
              src={Array []}
              unloader={false}
            />
            <div
              className="shop__item-title"
            >
              sdgf
            </div>
            <div
              className="shop__item-country"
            />
            <div
              className="shop__item-price"
            />
          </Link>,
          "className": "shop__item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Img
                alt="coffee"
                container={[Function]}
                decode={true}
                loader={false}
                src={Array []}
                unloader={false}
              />,
              <div
                className="shop__item-title"
              >
                sdgf
              </div>,
              <div
                className="shop__item-country"
              />,
              <div
                className="shop__item-price"
              />,
            ],
            "replace": false,
            "to": "/details/sdgf",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "coffee",
                "container": [Function],
                "decode": true,
                "loader": false,
                "src": Array [],
                "unloader": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "sdgf",
                "className": "shop__item-title",
              },
              "ref": null,
              "rendered": "sdgf",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
                "className": "shop__item-country",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "shop__item-price",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": [Function],
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          null,
          Array [
            <li
              className="shop__item"
              onClick={[Function]}
            >
              <Link
                replace={false}
                to="/details/wqw1"
              >
                <Img
                  alt="coffee"
                  container={[Function]}
                  decode={true}
                  loader={false}
                  src={Array []}
                  unloader={false}
                />
                <div
                  className="shop__item-title"
                >
                  wqw1
                </div>
                <div
                  className="shop__item-country"
                />
                <div
                  className="shop__item-price"
                />
              </Link>
            </li>,
            <li
              className="shop__item"
              onClick={[Function]}
            >
              <Link
                replace={false}
                to="/details/wqw2"
              >
                <Img
                  alt="coffee"
                  container={[Function]}
                  decode={true}
                  loader={false}
                  src={Array []}
                  unloader={false}
                />
                <div
                  className="shop__item-title"
                >
                  wqw2
                </div>
                <div
                  className="shop__item-country"
                />
                <div
                  className="shop__item-price"
                />
              </Link>
            </li>,
            <li
              className="shop__item"
              onClick={[Function]}
            >
              <Link
                replace={false}
                to="/details/sdgf"
              >
                <Img
                  alt="coffee"
                  container={[Function]}
                  decode={true}
                  loader={false}
                  src={Array []}
                  unloader={false}
                />
                <div
                  className="shop__item-title"
                >
                  sdgf
                </div>
                <div
                  className="shop__item-country"
                />
                <div
                  className="shop__item-price"
                />
              </Link>
            </li>,
          ],
        ],
        "className": "shop__wrapper",
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
        Object {
          "instance": null,
          "key": "wqw1",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/details/wqw1"
            >
              <Img
                alt="coffee"
                container={[Function]}
                decode={true}
                loader={false}
                src={Array []}
                unloader={false}
              />
              <div
                className="shop__item-title"
              >
                wqw1
              </div>
              <div
                className="shop__item-country"
              />
              <div
                className="shop__item-price"
              />
            </Link>,
            "className": "shop__item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Img
                  alt="coffee"
                  container={[Function]}
                  decode={true}
                  loader={false}
                  src={Array []}
                  unloader={false}
                />,
                <div
                  className="shop__item-title"
                >
                  wqw1
                </div>,
                <div
                  className="shop__item-country"
                />,
                <div
                  className="shop__item-price"
                />,
              ],
              "replace": false,
              "to": "/details/wqw1",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "alt": "coffee",
                  "container": [Function],
                  "decode": true,
                  "loader": false,
                  "src": Array [],
                  "unloader": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "wqw1",
                  "className": "shop__item-title",
                },
                "ref": null,
                "rendered": "wqw1",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": null,
                  "className": "shop__item-country",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "shop__item-price",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "wqw2",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/details/wqw2"
            >
              <Img
                alt="coffee"
                container={[Function]}
                decode={true}
                loader={false}
                src={Array []}
                unloader={false}
              />
              <div
                className="shop__item-title"
              >
                wqw2
              </div>
              <div
                className="shop__item-country"
              />
              <div
                className="shop__item-price"
              />
            </Link>,
            "className": "shop__item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Img
                  alt="coffee"
                  container={[Function]}
                  decode={true}
                  loader={false}
                  src={Array []}
                  unloader={false}
                />,
                <div
                  className="shop__item-title"
                >
                  wqw2
                </div>,
                <div
                  className="shop__item-country"
                />,
                <div
                  className="shop__item-price"
                />,
              ],
              "replace": false,
              "to": "/details/wqw2",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "alt": "coffee",
                  "container": [Function],
                  "decode": true,
                  "loader": false,
                  "src": Array [],
                  "unloader": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "wqw2",
                  "className": "shop__item-title",
                },
                "ref": null,
                "rendered": "wqw2",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": null,
                  "className": "shop__item-country",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "shop__item-price",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "sdgf",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/details/sdgf"
            >
              <Img
                alt="coffee"
                container={[Function]}
                decode={true}
                loader={false}
                src={Array []}
                unloader={false}
              />
              <div
                className="shop__item-title"
              >
                sdgf
              </div>
              <div
                className="shop__item-country"
              />
              <div
                className="shop__item-price"
              />
            </Link>,
            "className": "shop__item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Img
                  alt="coffee"
                  container={[Function]}
                  decode={true}
                  loader={false}
                  src={Array []}
                  unloader={false}
                />,
                <div
                  className="shop__item-title"
                >
                  sdgf
                </div>,
                <div
                  className="shop__item-country"
                />,
                <div
                  className="shop__item-price"
                />,
              ],
              "replace": false,
              "to": "/details/sdgf",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "alt": "coffee",
                  "container": [Function],
                  "decode": true,
                  "loader": false,
                  "src": Array [],
                  "unloader": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "sdgf",
                  "className": "shop__item-title",
                },
                "ref": null,
                "rendered": "sdgf",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": null,
                  "className": "shop__item-country",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "shop__item-price",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
